### PR TITLE
Conga line! Added support for dragging things through teleporters.

### PR DESCRIPTION
**Pull request by AvenDonn**
### Purpose
* Intuitive teleportation when dragging players or objects - They teleport with you.
* Simplify gameplay - Having to do the lag-dance on the teleporter to have the dragged entity teleport first, then teleport after it, is silly and doesn't always work.

### Notes:
* All entities are teleported to the same tile - Not trivial to fix.
* All entities are teleported at once - A 100-person conga line will squish into the target tile Dwarf Fortress party style.
* Dragging is not maintained through the teleporter for clients, probably due to update tick noticing something wrong.
    * Works for the host because magic
    * Can be handwaved as "It's hard to hold on while teleporting"
* QauntumPad, WorldGateway, and StationGateway had their transport code refactored into one TransportUtility class.
* Transporting reworked to use DisappearFromWorldServer and AppearAtPositionServer to perform the move without lerping

### Demonstration:
708c62a690ac3cabada05f6b12dba09d

### CODE REVIEW

File: UnityProject/Assets/Scripts/Gateway/StationGateway.cs [hunk 1]
```diff
@@ -26,8 +27,16 @@ public class StationGateway : NetworkBehaviour, IAPCPowered
 	[SerializeField]
 	private Sprite[] PowerOff = null;
 
+	/// <summary>
+	/// 
+	/// </summary>
 	private WorldGateway selectedWorld;// The world from the list that was chosen
```
Comment (AvenDonn) [inline UnityProject/Assets/Scripts/Gateway/StationGateway.cs:33]: Whoops, looks like I accidentally added xmldoc tags to this.

Comment (AvenDonn) [reply UnityProject/Assets/Scripts/Gateway/StationGateway.cs:30]: Removed.

**Comment by NoooneyDude** [comment]
Looks good.

**Comment by corp-0** [comment]
Hey, when you have some time can you please fix this:
```cs
if (pulledObj != null)
{
    TransportObject(pulledObj, transportTo);
    //Try to pull it again
    currentObj?.CmdPullObject(pulledObj?.gameObject);
    //TODO: Make pulling acutally continue working across teleporters for clients, not just server
    //TODO: Find a way to make the teleporter the teleport not all be on the same tile.
}
```

Using ``?`` operator in objects derived from GameObject might end in some Unity shenanigans. Sorry for that, I missed it when I quick read your pr.

**Comment by AvenDonn** [comment]
> Using `?` operator in derived from GameObject might end in some Unity shenanigans. Sorry for that, I missed it when I quick read your pr.

@corp-0 Huh, well, I learned something new about Unity's problems today I guess. I wasn't convinced, but I Googled it and I must say I am shocked and horrified by how Unity internally handles these things. 

I'll fix it later this week.

**Comment by AvenDonn** [comment]
@corp-0 as requested, #5065 has been opened

**Pull request merged**